src/common/decorators/swagger.ts

```diff
@@ -47,7 +49,59 @@ export const ApiArrayResponse = <TModel extends Type | string>(
     extraModels.push(model);
     items = { $ref: getSchemaPath(model) };
   }
+
+  if (Object.keys(extendItems).length > 0) {
```

Comment:
I think it would be better to make another decorator for this use case and hard code stuff for the claim, rather than introduce this complexity into a decorator that is widely used that didn't really need this stuff before.

I find its rather hard to follow what is happening here, it doesn't seem like a simplification from what was in a previous commit to me.

Reply:
I know this works at the moment, but my idea was to make the such a decorator that could handle ClaimModel<T> only. The change that you have made makes it workable with ApiArrayResponse but if we have to make it work individually that wouldn't work. 

Agreeing with Eric here, even if we aren't able to find an optimal solution we should separate this into a separate decorator and use it for our specific instance keeping the ApiArrayResponse untouched.

Reply:
I'll try to take a shot at it from my side and see if I can come up with something, until then I would like you to wait